pkg/pillar/zboot/zboot.go

```diff
@@ -332,11 +340,45 @@ func WriteToPartition(log *base.LogObject, srcFilename string, partName string)
 		return errors.New(errStr)
 	}
 
-	log.Infof("WriteToPartition %s, %s: %v\n", partName, devName, srcFilename)
+	log.Infof("WriteToPartition %s, %s: %v\n", partName, devName, image)
 
 	zbootMutex.Lock()
-	_, err := base.Exec(log, "dd", "if="+srcFilename, "of="+devName, "bs=8M").Output()
-	zbootMutex.Unlock()
+	defer zbootMutex.Unlock()
+
+	// use the edge-containers library to extract the data we need
+	puller := registry.Puller{
+		Image: image,
+	}
+	if casClient, err = cas.NewCAS(casClientType); err != nil {
```

Comment:
Why do you do this inside the mutex? Is sufficient to hold the mutex across the Pull() call to prevent concurrent modifications and reads of the partition.

Reply:
Nice catch. Addressed.